Sources/swiftML/Helpers.swift

```diff
@@ -0,0 +1,80 @@
+import TensorFlow
+
+/// The generalized inverse of a matrix using its singular-value decomposition (SVD).
+///
+/// Return the (Moore-Penrose) pseudo-inverse of a matrix.
+///
+/// - Parameter x: The matrix to pseudo-inverse.
+/// - Returns: Pseudo-inverse matrix.
+public func matrixPseudoInverse(_ data: Tensor<Float>) -> Tensor<Float> {
+    let svd = Raw.svd(data)
+    let diag = Raw.diag(diagonal: svd.s)
+    return matmul(matmul(svd.v, Raw.matrixInverse(diag)), svd.u.transposed())
+}
+
+/// Return Sign correction to ensure deterministic output from SVD.
+///
+/// - Parameters
+///   - u: The tensor containing of left singular vectors for each matrix.
+///   - v: The tensor containing of right singular vectors for each matrix.
+/// - Returns: Sign correction to ensure deterministic output.
+public func svdFlip(
+    u: Tensor<Float>,
+    v: Tensor<Float>,
+    uBasedDecision: Bool = true
+) -> (Tensor<Float>, Tensor<Float>) {
+  
+    var U = u
+    var V = v
+    
+    if uBasedDecision {
+    
+        let maxAbsCols = abs(u).argmax(squeezingAxis: 0)
+        var colValueForSign = Tensor<Float>(zeros: [u.shape[1]])
+        
+        for i in 0..<u.shape[1] {
+            colValueForSign[i] = u[Int(maxAbsCols[i].scalarized()), i]
+        }
+        
+        let signs = Raw.sign(colValueForSign)
+        U = U * signs
+        V = V * signs.reshaped(to: [u.shape[1], 1])
+        
+    } else {
+        
+        let maxAbsRows = abs(v).argmax(squeezingAxis: 1)
+        var rowValueForSign = Tensor<Float>(zeros: [u.shape[0]])
+        
+        for i in 0..<u.shape[0] {
+            rowValueForSign[i] = v[i, Int(maxAbsRows[i].scalarized())]
+        }
+        
+        let signs = Raw.sign(rowValueForSign)
+        U = U * signs
+        V = V * signs.reshaped(to: [u.shape[1], 1])
+    }
+    
+    return (U, V)
+}
+
+  
+/// Return the minkowski distance based on value of p.
```

Comment:
Documentation comment suggestions:
```suggestion
/// Returns the Minkowski distance between two tensors for the given distance metric `p`.
/// - Parameters:
///   - a: The first tensor.
///   - b: The second tensor.
///   - p: The order of the norm of the difference: `||a - b||_p`.
```

I don't feel a `/// - Returns` comment is necessary because the primary doc comment starts with `/// Returns`.

Reply:
I am thinking of specifying the data type/tensor structure in ' /// - Parameters' for type safety. What do you think about that?

Reply:
hey @dan-zheng,
For the documentation point of view, I was thinking of using [https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_markup_formatting_ref/index.html#//apple_ref/doc/uid/TP40016497-CH2-SW1](https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_markup_formatting_ref/index.html#//apple_ref/doc/uid/TP40016497-CH2-SW1). Is it ok? Or you have some specific guideline in mind?

Reply:
You don't need to specify types in `- Parameter:` comments, since Swift is already statically and strongly typed.

Reply:
yeah, Before seeing documentation I was thinking of that but now I don't.